generators/client-2/templates/src/main/webapp/app/_app.main.ts

```diff
@@ -1,4 +1,11 @@
+import { UpgradeAdapterRef } from "@angular/upgrade";
 import { upgradeAdapter } from './upgrade_adapter';
 import './app.module';
 
-upgradeAdapter.bootstrap(document.body, ['<%=angularAppName%>.app'], {strictDi: true});
+function readyFn(ref: UpgradeAdapterRef) {
+    let $urlRouter = ref.ng1Injector.get('$urlRouter');
+    $urlRouter.listen();
+    $urlRouter.sync();
+}
+
+upgradeAdapter.bootstrap(document.body, ['<%=angularAppName%>.app'], {strictDi: true}).ready(readyFn);
```

Comment:
I think I never figured this part when i tried and that might have made things problematic